debugger/src/gdb_scripts/linked_list_things.py

```diff
@@ -41,14 +151,31 @@ def invoke(self, arg, from_tty):
         #   # what address is being freed
         #   # look for the address in heap dictionary
 
+        # Extract stack frame information (state of local variables)
+        # and send to frontend client
+        # self.extract_stack_data()
+
+        send_heap_dict_to_server(self.user_socket_id, self.heap_dict)
+
         if any(t.is_running() for t in gdb.selected_inferior().threads()):
             gdb.execute('next')
 
         return self.heap_dict
 
+    def extract_stack_data(self):
```

Comment:
Could this be deleted now.
Or in a clean up branch, maybe.

Reply:
yes, deleted

Reply:
Just going to note this down though, this code by copilot does seem to work. It might lead us to more potentially useful gdb commands:

```python
def extract_stack_data(self):

        #  TODO: validate this copilot autogen
        # Extract stack frame information (state of local variables)
        # and send to frontend client
        print("=== Extracting stack frame information ===")
        frame = gdb.selected_frame()
        block = frame.block()
        for symbol in block:
            if symbol.is_argument or symbol.is_variable:
                print(symbol.name)
                print(symbol.value(frame))

## output:
'''
structssh-debugger-1  | val
structssh-debugger-1  | 11
structssh-debugger-1  | new
structssh-debugger-1  | 0xffff81898040 <_rtld_global>
'''
```